lib/WiFiNative/src/WifiClient.cpp

```diff
@@ -0,0 +1,162 @@
+/* MIT License
+ *
+ * Copyright (c) 2023 - 2024 Andreas Merkle <web@blue-andi.de>
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+/*******************************************************************************
+    DESCRIPTION
+*******************************************************************************/
+/**
+ *  @brief WifiClient class for ArduinoNative.
+ *  @author Norbert Schulz <github@schulznorbert.de>
+ */
+
+/******************************************************************************
+ * Includes
+ *****************************************************************************/
+#include "WiFiClient.h"
+
+#include <unistd.h>
+#include <sys/socket.h>
+#include <sys/fcntl.h>
+#include <netinet/tcp.h>
+#include <netinet/in.h>
+
+/******************************************************************************
+ * Macros
+ *****************************************************************************/
+
+/******************************************************************************
+ * Prototypes
+ *****************************************************************************/
+
+/******************************************************************************
+ * Local Variables
+ *****************************************************************************/
+
+
+/******************************************************************************
+ * Public Methods
+ *****************************************************************************/
+
+uint8_t WiFiClient::connect(const IPAddress& addr, uint16_t port)
+{
+    uint8_t retval = 0;
+
+    if (connected())
+    {
+        stop();
```

Comment:
Why closing the connection?

Reply:
Ah I see the intention now ... we abort a running connection, even if we are already connected to the same remote device.